src/core/namespace.c

```diff
@@ -857,10 +860,50 @@ static bool namespace_info_mount_apivfs(const char *root_directory, const NameSp
          */
 
         /* root_directory should point to a mount point */
-        return root_directory &&
-                (ns_info->mount_apivfs ||
-                 ns_info->protect_control_groups ||
-                 ns_info->protect_kernel_tunables);
+        if (!root_directory)
+                return 0;
+
+        if (ns_info->mount_apivfs)
+                return ELEMENTSOF(apivfs_table);
+
+        /* Count /sys mount but only inside RootDirectory= RootImage= */
+        if (ns_info->protect_control_groups)
+                cnt++;
+
+        /* Count /proc and /sys mounts but only inside RootDirectry= RootImage= */
+        if (ns_info->protect_kernel_tunables)
+                cnt += 2;
```

Comment:
if both ProtectControlgroups= and ProtectKernelTunables= is set you'd count /sys twice here, no?